### PR TITLE
Update README instructions

### PR DESCRIPTION
**Pull request by anvacaru**
closes: #150 
Add instructions on how to build from source and update the expected output of the tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,58 @@
 # Kontrol
 
+Fast Installation
+-----------------
 
-## Installation
+-   `bash <(curl https://kframework.org/install)`: install [kup package manager].
```
Comment (palinatolmach) [inline README.md:6]:
```suggestion
-   `bash <(curl https://kframework.org/install)`: install [kup package manager](https://github.com/runtimeverification/kup).
```

File: README.md [hunk 2]
```diff
@@ -1,17 +1,58 @@
...
+**NOTE**: The first run will take longer to fetch all the libraries and compile sources. (30m to 1h)
 
-```bash
-make build
-pip install dist/*.whl
+### Build from source
+
+#### K Framework
+
+You need to install the [K Framework] on your system, see the instructions there.
```
Comment (palinatolmach) [inline README.md:16]:
```suggestion
You need to install the [K Framework](https://github.com/runtimeverification/k) on your system, see the instructions there.
```

File: README.md [hunk 3]
```diff
@@ -1,17 +1,58 @@
...
+The fastest way is via the [kup package manager], with which you can do to get the correct version of K:
```
Comment (palinatolmach) [inline README.md:17]:
```suggestion
The fastest way is via the [kup package manager](https://github.com/runtimeverification/kup), with which you can do to get the correct version of K:
```

File: README.md [hunk 4]
```diff
@@ -1,17 +1,58 @@
...
 ```
 
+To change the default compiler:
+```sh
+CXX=clang++-14 poetry run kevm-dist --verbose build -j`nproc` plugin haskell foundryx
+```
+
+On Apple Silicon:
+```sh
+APPLE_SILICON=true poetry run kevm-dist --verbose build -j`nproc` plugin haskell foundryx
```
Comment (palinatolmach) [inline README.md:44]: `nproc` is not a command on Mac; apparently the equivalent is `sysctl -n hw.logicalcpu`, but maybe we can also recommend some fixed number of processes, e.g., `-j3`.

**Review by palinatolmach — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (anvacaru) [reply README.md:5]: Added the link at the end of the document to avoid duplication.

**Pull request merged**